internal/pkg/sylog/sylog.go

```diff
@@ -205,3 +206,21 @@ func Writer() io.Writer {
 
 	return os.Stderr
 }
+
+// DebugLogger is an implementation of the go-log/log Logger interface that will
+// output log messages via sylog.debug when required by external packages such
+// as the scs-library-client
+type DebugLogger struct {
+}
+
+var _ golog.Logger = (*DebugLogger)(nil)
+
+// Output a log message via sylog.Debugf
+func (t DebugLogger) Log(v ...interface{}) {
+	Debugf("%s", fmt.Sprint(v...))
```

Comment:
By using `Debugf` here sylog will always report `Log`/`Logf` as the caller in trace because of https://github.com/sylabs/singularity/blob/2206b6d1fcacf25ad6a6c82bd54166ab5bc2d8e7/internal/pkg/sylog/sylog.go#L93
Replacing by `writef(os.Stderr, debug, format, a...)` should fix that

Reply:
Yeah - I just noticed that, will fix.